examples/web/cache/refresh-disposable-tokens.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  AllTopics,
+  AuthClient,
+  CredentialProvider,
+  DisposableTokenScopes,
+  ExpiresAt,
+  ExpiresIn,
+  GenerateDisposableTokenResponse,
+  TopicClient,
+  TopicItem,
+  TopicPublish,
+  TopicPublishResponse,
+  TopicSubscribe,
+  TopicSubscribeResponse,
+} from '@gomomento/sdk-web';
+import {initJSDom} from './utils/jsdom';
+
+// In your own setup, the token vending machine would likely be a separate service.
+// In this file, we provide a local version if you want to run this example completely independently.
+// We also provide a version that fetches a token from a deployed token vending machine.
+// See https://github.com/momentohq/client-sdk-javascript/tree/main/examples/nodejs/token-vending-machine
+// for more information about deploying your own token vending machine.
+
+async function localTokenVendingMachine(): Promise<{token: string; expiresAt: ExpiresAt}> {
+  const authClient = new AuthClient({});
+  const tokenResponse = await authClient.generateDisposableToken(
+    DisposableTokenScopes.topicPublishSubscribe('my-cache', AllTopics),
+    ExpiresIn.seconds(30)
+  );
+  if (tokenResponse.type === GenerateDisposableTokenResponse.Error) {
+    throw new Error(`Failed to generate a disposable token: ${tokenResponse.toString()}`);
+  }
+  const humanReadableTime = new Date(tokenResponse.expiresAt.epoch() * 1000).toISOString();
+  console.log(`Generated a disposable token that will expire at ${humanReadableTime}`);
+  const disposableToken = {
+    token: tokenResponse.authToken,
+    expiresAt: tokenResponse.expiresAt,
+  };
+  return disposableToken;
+}
+
+async function tokenVendingMachine(): Promise<{token: string; expiresAt: ExpiresAt}> {
```

Comment:
@poppoerika I'm late reviewing this PR but *this* is the function that they would implement on their ipads. It will just call `fetch` with whatever URL, headers, query params are necessary to get a token from their server-side token vending endpoint.

Their server-side token-vending endpoint will be the thing that uses AuthClient to actually get the tokens.